test/test_filo.c

```diff
@@ -45,19 +44,47 @@ int main(int argc, char* argv[])
   const char* filelist[1] = { filename };
   const char* dest_filelist[1] = { dest_filename };
 
+  /* base path for storage is NULL, so destination files will be written to the local dir*/
   rc = Filo_Flush("mapfile", NULL, 1, filelist, dest_filelist, MPI_COMM_WORLD);
+//printf("pre-fetch src file name is %s\n", filelist[0]);
+//printf("pre-fetch dst_file name is %s\n", dest_filelist[0]);
+////in file name, rank precedes ".out" suffix
+//int rank_from_file_name = *((strstr(dest_filelist[0], ".out"))-1) - '0';
+//printf("char at .out is %d\n",rank_from_file_name);
 
   unlink(filename);
 
   /* fetch list of files recorded in mapfile to /dev/shm */
   int num_files;
   char** src_filelist;
   char** dst_filelist;
+  /* src base path is still NULL (consistent with Filo_Flush), but the dest base path is /dev/shm*/
   rc = Filo_Fetch("mapfile", NULL, "/dev/shm", &num_files, &src_filelist, &dst_filelist, MPI_COMM_WORLD);
 
   /* free file list returned by fetch */
   int i;
   for (i = 0; i < num_files; i++) {
+    //in file name, rank precedes ".out" suffix
+    int rank_from_file_name = *((strstr(dst_filelist[i], ".out"))-1) - '0';
+    //assertain that the filename with consistant process marker was passed through flush/fetch
+    if(rank != rank_from_file_name){
+      rc = TEST_FAIL;
+      printf("rank = %d, rank_from_file_name = %d\n", rank, rank_from_file_name);
+    }
+    //assertain that the file content is consistent with the process
+    FILE *file = fopen(filename, "r");
```

Comment:
Just to be certain, let's change this to fopen ```dst_filelist[i]``` instead of ```filename```.

Go ahead and close the file too after reading its contents for file handle cleanup.